### PR TITLE
Show dialog when user denies location permission

### LINKED ISSUE #241: Should we show the user an info dialog after they deny enabling location
Currently, when a user clicks on the location button and they have not enabled GPS(location) a dialog is shown asking them if they want to turn it on. In case they deny, the dialog is dismissed and the user is not focused onto their location.

I am not sure if the users are clever enough to know that the reason the map is not focusing on their location is because they denied enabling location. Should we show an info dialog after they deny enabling location?

This is the current dialog shown. Does it give enough information to the user? Should we add another in case the user denies enabling the location(GPS)?

screenshot_20190117-105352
screenshot_20190117-105358

### PR DESCRIPTION
**Pull request by ekigamba**
- If the user is prompted for location permissions and rejects it, a dialog telling him/her that s/he can no longer use the My Location feature will be shown

Fixes #241

### CODE REVIEW

File: library/src/main/java/io/ona/kujaku/views/KujakuMapView.java [hunk 1]
```diff
@@ -1085,6 +1088,31 @@ public void onResume() {
         if (warmGps && Permissions.check(getContext(), Manifest.permission.ACCESS_FINE_LOCATION)) {
             checkLocationSettingsAndStartLocationServices(true);
         }
+
+        // Explain the consequence of rejecting enabling location
+        if (isRequestingEnableLocation) {
```
Comment (vincent-karuri) [inline library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: Why do you need `isRequestingEnableLocation`

Can't `hasAlreadyRequestedEnableLocation` be used?

**Comment by vincent-karuri** [comment]
Can most of this functionality be moved to the LocationSettingsHelper?

KujakuMapView is growing big and it might be good to split functionality to helpers.

File: library/src/main/java/io/ona/kujaku/views/KujakuMapView.java (hunk 1 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: They don't store the same state. They will both have(and should have) different states when `onResume` is called

**Comment by ekigamba** [comment]
> Can most of this functionality be moved to the LocationSettingsHelper?
> 
> KujakuMapView is growing big and it might be good to split functionality to helpers.

I currently don't see any of the added code blocks that can be candidates for that. Could you point them out if any?

**Comment by vincent-karuri** [comment]
> > Can most of this functionality be moved to the LocationSettingsHelper?
> > KujakuMapView is growing big and it might be good to split functionality to helpers.
> 
> I currently don't see any of the added code blocks that can be candidates for that. Could you point them out if any?

A possibility is you can have a new method `checkLocationEnabledWithMessage` that will call the existing `checkLocationEnabled` and build the AlertDialog there. 

Or some other approach.

File: library/src/main/java/io/ona/kujaku/views/KujakuMapView.java (hunk 1 above)
Comment (vincent-karuri) [reply library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: It's not clear what each does then.

Maybe add comments at the top where the variables are declared to delineate this.

Comment (vincent-karuri) [reply library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: Great!

**Pull request merged**